apps/api/src/domain/models/media-item/commands/publish-media-item.command.integration.spec.ts

```diff
@@ -0,0 +1,193 @@
+
+    let commandHandlerService: CommandHandlerService;
+
+    let arangoConnectionProvider: ArangoConnectionProvider;
+
+    let idManager: IIdManager;
+
+    let assertionHelperDependencies: Omit<CommandAssertionDependencies, 'idManager'>;
+
+    beforeAll(async () => {
+        ({ testRepositoryProvider, commandHandlerService, idManager, arangoConnectionProvider } =
+            await setUpIntegrationTest({
+                ARANGO_DB_NAME: 'testonly-333',
+                // generateRandomTestDatabaseName(),
+            }));
+
+        commandHandlerService.registerHandler(
+            commandType,
+            new PublishMediaItemCommandHandler(
+                testRepositoryProvider,
+                idManager,
+                ResourceType.mediaItem
+            )
+        );
+
+        assertionHelperDependencies = {
+            testRepositoryProvider,
+            commandHandlerService,
+        };
+    });
+
+    afterAll(async () => {
+        await arangoConnectionProvider.dropDatabaseIfExists();
+    });
+
+    beforeEach(async () => {
+        await testRepositoryProvider.testSetup();
+    });
+
+    afterEach(async () => {
+        await testRepositoryProvider.testTeardown();
+    });
+
+    describe('when the command is valid', () => {
+        it('should succeed', async () => {
+            await assertCommandSuccess(assertionHelperDependencies, {
+                buildValidCommandFSA: () => validCommandFSA,
+                initialState,
+                checkStateOnSuccess: async () => {
+                    const result = await testRepositoryProvider
+                        .forResource(ResourceType.mediaItem)
+                        .fetchById(validCommandFSA.payload.id);
+
+                    expect(result).toBeInstanceOf(MediaItem);
+
+                    const updatedInstance = result as MediaItem;
+
+                    const { id, published } = updatedInstance;
+
+                    expect(id).toBe(unpublishedMediaItem.id);
+
+                    expect(published).toBe(true);
+
+                    expect(
+                        updatedInstance.eventHistory.some(
+                            ({ type }) => type === 'MEDIA_ITEM_CREATED'
+                        )
+                    );
+                },
+            });
+        });
+    });
+
+    describe('when the command payload has an invalid type', () => {
+        describe('when the id is missing', () => {
+            it('should fail', async () => {
+                await assertCommandError(assertionHelperDependencies, {
+                    buildCommandFSA: () => ({
+                        type: commandType,
+                        payload: {},
+                    }),
+                    initialState,
+                    checkError: (error: InternalError) => {
+                        expect(error).toBeInstanceOf(InvalidCommandPayloadTypeError);
+
+                        expect(error.innerErrors[0].toString().includes('id')).toBe(true);
```

Comment:
Export a helper from the `CREATE_MEDIA_ITEM` test suite for checking type errors and use it here.

Reply:
done